dev/Dockerfile-test

```diff
@@ -1,8 +1,89 @@
-FROM python:3.9
+# `python-base` sets up all our shared environment variables
+FROM python:3.9-slim as python-base
 
-RUN pip install requests
+ENV PYTHONUNBUFFERED=1 \
+    # prevents python creating .pyc files
+    PYTHONDONTWRITEBYTECODE=1 \
+    \
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100 \
+    \
+    # https://python-poetry.org/docs/configuration/#using-environment-variables
+    POETRY_VERSION=1.1.5 \
+    # make poetry install to this location
+    POETRY_HOME="/opt/poetry" \
+    # make poetry create the virtual environment in the project's root
+    # it gets named `.venv`
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    # do not ask any interactive question
+    POETRY_NO_INTERACTION=1 \
+    \
+    # this is where our requirements + virtual environment will live
+    PYSETUP_PATH="/opt/pysetup" \
+    VENV_PATH="/opt/pysetup/.venv"
 
-# COPY ./run_e2e_tests.sh ./
-COPY ./run_session.py ./
+# prepend poetry and venv to path
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
 
-CMD [ "" ]
+# `development` image is used during development / testing
+FROM python-base as development
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV GECKODRIVER_VER v0.30.0
+ENV FIREFOX_VER 93.0
+
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        curl \
+        build-essential \
+        firefox-esr \
```

Comment:
Is this using firefox-esr? I see that 93 is downloaded below...

Reply:
Good catch. I'd missed cleanup in that Dockerfile.